src/content/development/contribution-guide/_index.en.md

```diff
@@ -234,6 +235,16 @@ During the week when the general availability release is performed, the next ver
 Additionally, a retrospective meeting for the last release cycle will be held.
 There will be no dedicated [test day](#test-days), as the release candidate has been tested and no changes have occurred since.
 
+##### Release Notes
+
+[Release notes] should be published for each GA version.
+All new features and bug fixes of the release should be labeled with `release-note-needed` and have a relevant release note
+documented as part of the PR. The content of the release-note should be in a file under `docs/release-notes`.
```

Comment:
I've put recent release notes simply under `release-notes` directory.
If we go with `docs/release-notes` we'll need a PR to move them.

Reply:
Just `release-notes` seems good to me, as we're not planning on putting any other docs there.